### PR TITLE
test: Extract assertion logic from the helpers.js file and add it to an Assertions class.

### PR DESCRIPTION
**Pull request by cortisiko**
## **Description**
Problem

To address the issue of the helpers.js file containing numerous test actions, and assertions, which can be confusing for newcomers to the framework, one possible solution is to refactor the code by extracting the assertions into a separate assertion class. This will make it easier to identify and use the assertions within e2e the tests.

Solution

Create a new file called assertions.js or any appropriate name for the assertion class. The goal here is to remove the assertions in these lines of code: https://github.com/MetaMask/metamask-mobile/blob/main/e2e/helpers.js#L121#L164 to its own assertion class.

In the assertions.js file, define a class that contains reusable assertion methods.

Important: The introduced code will remain unused. This pull request is 3 of 3 PRs, which involves cleaning up the helpers.js file.

## **Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change._

### **Before**

_[screenshot]_

### **After**

_[screenshot]_

## **Related issues**

_Fixes #???_

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by SamuelSalas — DISMISSED** [review]

**Comment by SamuelSalas** [comment]
LGTM

File: e2e/utils/Assertions.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+const TIMEOUT = 15000;
+
+/**
+ * Class representing a set of assertions for Detox testing.
+ */
+class Assertions {
+  /**
+   * Check if an element with the specified ID is visible.
+   * @param {string} elementId - The ID of the element to check.
+   * @param {number}  Timeout in milliseconds.
```
Comment (leotm) [inline e2e/utils/Assertions.js:13]:
```suggestion
   * @param {number} Timeout in milliseconds.
```

File: e2e/utils/Assertions.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    // Rename me. The naming convention here is terrible.
+
+    return await waitFor(elementId).not.toHaveText(text).withTimeout(timeout);
+  }
+
+  static async checkIfExists(elementId, timeout = TIMEOUT) {
+    /* This assertion seems a bit redudant. Please replace me with checkIfVisible
+     and delete me
+    We already have an assertion to check if an element is visible.
+    No need to check if it exists
```
Comment (leotm) [inline e2e/utils/Assertions.js:58]:
```suggestion
    No need to check if it exists.
```

File: e2e/utils/Assertions.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      .toBeVisible()
+      .withTimeout(timeout);
+    return expect(element(by.id(elementId))).toExist();
+  }
+
+  /**
+   * Check if an element with the specified ID has the specified text.
+   * @param {string} elementId - The ID of the element to check.
+   * @param {string} text - The expected text content.
+   * @param {number}  timeout - Timeout in milliseconds.
```
Comment (leotm) [inline e2e/utils/Assertions.js:71]:
```suggestion
   * @param {number} timeout - Timeout in milliseconds.
```

**Review by leotm — DISMISSED** [review]
lgtm

**Comment by cortisiko** [comment]
added your suggestions @leotm ! please review it one more time.

**Review by leotm — DISMISSED** [review]

**Review by chrisleewilcox — COMMENTED** [review]
Can we update one test file that uses one of the assertions in the new file to show how/it works?

**Comment by cortisiko** [comment]
@chrisleewilcox I didn't want to merge the example into this PR because it would have required time to restructure a few pages. I branched off this branch and created an example. See example here: https://github.com/MetaMask/metamask-mobile/compare/main...detox/example-pom#diff-1eb196708d872982bf1ba97d28e5c5a0fa11db2535e052c32e60139bf422044dR30

If it makes sense to you, can you review and approve this PR

**Review by chrisleewilcox — APPROVED** [review]

**Pull request merged**